presto-main/src/main/java/com/facebook/presto/type/MapType.java

```diff
@@ -69,6 +71,25 @@ public Type getValueType()
     }
 
     @Override
+    public boolean isComparable()
+    {
+        return valueType.isComparable();
+    }
+
+    @Override
+    public int hash(Block block, int position)
+    {
+        Slice mapSlice = getSlice(block, position);
+        Block mapBlock = readStructuralBlock(mapSlice);
+        Map<Integer, Integer> hashMap = new HashMap<>();
+
+        for (int i = 0; i < mapBlock.getPositionCount(); i += 2) {
+            hashMap.put(hashPosition(keyType, mapBlock, position), hashPosition(valueType, mapBlock, position + 1));
```

Comment:
How's this one:
hashMap.put(keyType.hash(mapBlock, position), valueType.hash(mapBlock, position + 1));